2a_model.R

```diff
@@ -101,6 +101,84 @@ p2a_targets_list <- list(
       subset_and_write_zarr(do_and_metab, "2a_model/out/well_observed_trn_val_targets.zarr", p2a_trn_val_sites)
     },
     format="file"
+  ),
+  
+  
+  ## CREATE EQUIVALENT TARGETS FOR "MODERATELY-OBSERVED SITES" ##
+  # char vector of moderately-observed train sites
+  tar_target(
+    p2a_trn_sites_medobs,
+    p2_med_observed_sites[!(p2_med_observed_sites %in% val_sites) & !(p2_med_observed_sites %in% tst_sites)]
+  ),
+  
+  # char vector of moderately-observed val and training sites
+  tar_target(
+    p2a_trn_val_sites_medobs,
+    p2_med_observed_sites[(p2_med_observed_sites %in% p2a_trn_sites_medobs) | (p2_med_observed_sites %in% val_sites)]
+  ),
+  
+  # get moderately-observed sites that we use for trning, but also have data in the val time period
+  tar_target(
+    p2a_trn_sites_w_val_data_medobs,
+    p2_daily_with_seg_ids  %>%
+      filter(site_id %in% p2a_trn_val_sites_medobs,
+             !site_id %in% val_sites,
+             date >= val_start_date,
+             date < val_end_date) %>%
+      group_by(site_id) %>%
+      summarise(val_count = sum(!is.na(do_mean))) %>%
+      filter(val_count > 0) %>%
+      pull(site_id)
+  ),
+  
+  # moderately-observed sites that are trning sites but do not have data in val period
+  tar_target(
+    p2a_trn_only_medobs,
+    p2a_trn_sites_medobs[!p2a_trn_sites_medobs %in% p2a_trn_sites_w_val_data_medobs]
+  ),
```

Comment:
These two targets are only used in making a sf object in 3_visualize:
https://github.com/USGS-R/drb-do-ml/blob/6e0afdcc0aed84b7aae1d3497775ff5835e12faf/3_visualize.R#L51-L62

So I'm not entirely sure we should have them (or the ones for the "well_obs" sites) in 2a. It may actually be better just to make them as part of the `p3_well_observed_site_data` target if we end up making that. But for now, I'd suggest leaving these out.

Reply:
OK, good to know. I omitted the equivalent of these targets since it makes the 2a_model phase file pretty bulky. Instead, I just filtered the met data for sites that are within `p2_med_observed_sites` and _not_ within the `tst_sites` vector defined in `_targets.R`.